pkgs/by-name/ha/haraka/package.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, python3
+}:
+
+buildNpmPackage rec {
+  pname = "haraka";
+  version = "3.0.2";
+
+  src = fetchFromGitHub {
+    owner = "haraka";
+    repo = "Haraka";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-JCrgKpqX+qS1P7WNlxq90iXysW7gRERD+UVWgz1U9II=";
+  };
+
+  npmDepsHash = "sha256-9LMkMmOJJ0cmq1hsoYW+SMTptffujk8lcWNX9pSVJrY=";
+
+  postPatch = ''
+    cp ${./package-lock.json} ./package-lock.json
```

Comment:
Hm, why do we need to copy it? Usually, the `buildNpmPackage` system requires only the `npmDepsHash`, is it because upstream does not have one?

Reply:
Exactly. The upstream people are [firm believers](https://github.com/haraka/Haraka/blob/master/.gitignore#L15) of no locking. 🤷

Reply:
*Sighs*, maybe it would be nice to have an auto-updater that refreshes the package-lock in that case. Otherwise, those sorts of packages tends to rot quite fast, unfortunately.

Also, you only need the package-lock to be present during the derivation, not in this repo, do you?
You can still have a magic patch you fetch and add the package-lock.

The reason for why I'm saying all of this is because this is adding 8kb to the repo and people will complain :P.

Reply:
Good point. I'll see what I can hack up together. 👍

Reply:
If it's too hard @farcaller, let me know and we can merge it like this. I won't die on the hill of blocking that PR for this.

Reply:
Apparently, the upstream isn't against the lockfile per se, there was a discussion in https://github.com/haraka/Haraka/issues/2960 on that specific issue. I filed https://github.com/haraka/Haraka/issues/3271 to see if they will reconsider.